### PR TITLE
Redact secrets using "+" rather than "*" as this is base 64 compatiba…

### LINKED ISSUE #1923: Warning during secret diffing
**Describe the bug**
Following warning is printed during secret diffing:

```
WARN[0022] Could not unmarshal to object of type /v1, Kind=Secret: illegal base64 data at input byte 0 
```

The warning pollutes `argocd app diff` output. The error happens during secret re-marshaling. This is not a problem now, but might be a problem in a future if secret get a field which require remarshaling .

**To Reproduce**
1. Deploy any app which has secret
2. Sync in and run `argocd app diff`
3. See warning message which makes it difficult to read command output

**Version**: v1.1.0

### PR DESCRIPTION
**Pull request by alexec**
…le. Closes #1923

### CODE REVIEW

**Comment by alexec** [comment]
Genuine failure:`TestAppWithSecrets`

File: test/e2e/app_management_test.go [hunk 1]
```diff
@@ -284,7 +284,7 @@ func TestAppWithSecrets(t *testing.T) {
 
 			diffOutput, err := fixture.RunCli("app", "diff", app.Name)
 			assert.Error(t, err)
-			assert.Contains(t, diffOutput, "username: '*********'")
+			assert.Contains(t, diffOutput, "username: +++++++++")
```
Comment (alexec) [inline test/e2e/app_management_test.go]: note the quoting change

**Comment by alexec** [comment]
I think build is being flakey. Re-running.

**Comment by alexec** [comment]
`TestAutoSyncSelfHealEnabled` and `TestHookDeleteBeforeCreation` failed. I don't think these are related. Rebuilding, but I have already re-built once.

**Comment by alexec** [comment]
`TestHookDeleteBeforeCreation` failed.

**Comment by alexec** [comment]
Fails on `TestHookDeleteBeforeCreation` - known flaky test.

**Review by alexmt — APPROVED** [review]

**Pull request merged**